sound/soc/sof/sof-client-probes.c

```diff
@@ -430,7 +416,22 @@ static int sof_probes_client_probe(struct auxiliary_device *auxdev,
 	}
 
 	priv->host_ops = ops;
-	priv->ipc_ops = &ipc3_probe_ops;
+
+	if (sof_client_get_ipc_type(cdev) == SOF_INTEL_IPC4) {
+#ifdef CONFIG_SND_SOC_SOF_INTEL_IPC4
```

Comment:
nit-pick: i prefer the use of if (IS_ENABLED(CONFIG_SND_SOC_SOF_INTEL_IPC4)) to avoid breaking the indentation. The compiler is smart-enough to prune always-false branches.

Reply:
can we change the if/ifelse to a switch case? will make it easier to add new IPCs later on

Reply:
Sure, I can use switch(sof_client_get_ipc_type(cdev)), but the references to IPC4 code must be filtered out by preprocessor  to avoid "undefined symbols".